src/qibocal/protocols/characterization/fast_reset/fast_reset.py

```diff
@@ -0,0 +1,308 @@
+from dataclasses import dataclass, field
+from typing import Dict, Optional
+
+import numpy as np
+import numpy.typing as npt
+import plotly.graph_objects as go
+from plotly.subplots import make_subplots
+from qibolab import ExecutionParameters
+from qibolab.platform import Platform
+from qibolab.pulses import PulseSequence
+from qibolab.qubits import QubitId
+
+from qibocal.auto.operation import Data, Parameters, Qubits, Results, Routine
+
+# TODO: IBM Fast Reset until saturation loop
+# https://quantum-computing.ibm.com/lab/docs/iql/manage/systems/reset/backend_reset
+
+
+@dataclass
+class FastResetParameters(Parameters):
+    """FastReset runcard inputs."""
+
+    nshots: Optional[int] = None
+    """Number of shots."""
+    relaxation_time: Optional[int] = None
+    """Relaxation time (ns)."""
+
+
+@dataclass
+class FastResetResults(Results):
+    """FastReset outputs."""
+
+    optimal_integration_weights: Dict[QubitId, float]
+    """
+    Optimal integration weights for a qubit given by amplifying the parts of the
+    signal acquired which maximally distinguish between state 1 and 0.
+    """
+
+
+FastResetType = np.dtype(
+    [
+        ("probability", np.float64),
+        ("state", np.int64),
+        ("iteration", np.int64),
+        ("fast_reset", np.int64),
+    ]
+)
+"""Custom dtype for FastReset."""
+
+
+@dataclass
+class FastResetData(Data):
+    """FastReset acquisition outputs."""
+
+    data: dict[QubitId, npt.NDArray[FastResetType]] = field(default_factory=dict)
```

Comment:
If I understood correctly each array is identified by 3 keys (qubit,state,fast_reset), right?
Potentially we could store each array by selecting these 3 keys:
```suggestion
    data: dict[tuple[QubitId,int,int], npt.NDArray[FastResetType]] = field(default_factory=dict)
```
This should also fix the issue related to the shape I think.
Moreover, you don't need to store multiple times the state and the fast_reset.

I did something similar for single_shot_classification (only using state):

https://github.com/qiboteam/qibocal/blob/8d2b63099486faab8355cc4fa92b59abc7213ab6/src/qibocal/protocols/characterization/classification.py#L35

Reply:
Yeah, that would be perfect I will change it the same way in #383